__tests__/app.test.js

```diff
@@ -653,6 +653,144 @@ describe('GET /api/users', () => {
     });
 });
 
+describe('GET /api/articles (queries)', () => {
+    test('200: should have 13 articles when unrecognised query is ignored', () => {
+        return request(app)
+        .get('/api/articles?not_a_variable=not_a_value')
+        .expect(200)
+        .then(({ body }) => {
+            expect(body.articles).toHaveLength(13);
+        });
+    });
+
+    test('200: should have 12 articles when filtered by topic "mitch"', () => {
+        return request(app)
+        .get('/api/articles?topic=mitch')
+        .expect(200)
+        .then(({ body }) => {
+            expect(body.articles).toHaveLength(12);
+
+            for (const article of body.articles) {
+                expect(article.topic).toBe('mitch');
+            }
+        });
+    });
+
+    test('200: should have 1 article when filtered by topic "CaTs" (when topic is not lowercase)', () => {
+        return request(app)
+        .get('/api/articles?topic=CaTs')
+        .expect(200)
+        .then(({ body }) => {
+            expect(body.articles).toHaveLength(1);
+            expect(body.articles[0].topic).toBe('cats');
+        });
+    });
+    
+    test('200: should have an empty array when filtered by topic "not_a_topic"', () => {
```

Comment:
I think an invalid topic should really give a 404, otherwise we could risk SQL injection